File: app/src/main/java/com/ternaryop/photoshelf/db/Blog.java
package com.ternaryop.photoshelf.db;

import java.io.Serializable;
import java.util.Locale;

import android.provider.BaseColumns;

public class Blog implements BaseColumns, Serializable {
    private static final long serialVersionUID = 5887665998065237345L;
    private long id;
    private String name;

    public Blog() {
    }

    public Blog(long id, String name) {
        this.id = id;
        this.name = name.toLowerCase(Locale.US);
    }

    public Blog(String name) {
        this(0, name);
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name.toLowerCase(Locale.US);
    }

    @Override
    public String toString() {
        return name;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }
}

File: app/src/main/java/com/ternaryop/photoshelf/db/Blog.kt
package com.ternaryop.photoshelf.db

import android.provider.BaseColumns
import java.io.Serializable
import java.util.Locale

class Blog(var id: Long, name: String) : BaseColumns, Serializable {
    var name: String = name.toLowerCase(Locale.US)
    set(value) {
        field = name.toLowerCase(Locale.US)
    }

    override fun toString(): String = name

    companion object {
        private const val serialVersionUID = 5887665998065237345L
    }
}